app/controllers/documents_controller.rb

```diff
@@ -1,7 +1,56 @@
+require 'gds_api/publishing_api_v2'
+
 class DocumentsController <  ApplicationController
 
   def index
     redirect_to "/#{document_types.keys.first}" unless params[:document_type]
   end
 
+  def new
+    render :new, locals: { document: current_format.klass.new }
+  end
+
+  def create
+    document = current_format.klass.new(
+      filtered_params(params[:"#{current_format.format_name}"])
```

Comment:
Since filtered_params returns a HashWithIndifferentAccess, you could just use a string key which would be a bit clearer.